### PR TITLE
"Collection" has no semantics

### LINKED ISSUE #147: Is a FeatureCollection a set or a sequence?
https://tools.ietf.org/html/draft-ietf-geojson-00#section-2.3 is very short and says that a FeatureCollection is an array. It does not say whether this array represents a set or a sequence, i.e. whether GeoJSON implementations need to be careful to maintain order when processing GeoJSON. maybe it would be helpful to be explicit about that?

### PR DESCRIPTION
**Pull request by sgillies**
A reframing of #151.

Closes #147.
Closes #151.

### CODE REVIEW

**Comment by dret** [comment]
i like this. should be referenced from #147 (and potentially close it), i guess.

**Comment by dret** [comment]
off-topic note: if this is accepted, maybe add the same language to geometry collection?

**Comment by mpdaly** [comment]
Or, perhaps better still, move it and reword it to cover both FeatureCollection and GeometryCollection, instead of repeating the text.

Also "a JSON array", not "an JSON array".

**Comment by dret** [comment]
On 2016-02-19 10:37, Martin Daly wrote:
> Or, perhaps better still, move it and reword it to cover both
> FeatureCollection and GeometryCollection, instead of repeating the text.

in that case this could become part of addressing #149, which probably 
should add some intro text to section 2, explaining how the concepts 
defined there relate.

as part of addressing $149, my goal is to propose some ascii art similar 
to 
https://github.com/geojson/draft-geojson/issues/148#issuecomment-185088766 
as an overview.

**Comment by mpdaly** [comment]
SFSQL is referenced, and has all the non-ascii art regarding geometry class hierarchy that anyone could ever want or need. We don't repeat the JSON syntax diagrams either.

**Comment by sgillies** [comment]
I've consolidated the explanation about "collection" into the definitions section.

**Comment by mpdaly** [comment]
:+1:
